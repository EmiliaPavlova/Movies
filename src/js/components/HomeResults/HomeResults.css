.rightSection {
  display: flex;
  justify-content: space-between;
  width: 20%;
  min-width: 180px;
}

.selected {
  color: #F55263;
}