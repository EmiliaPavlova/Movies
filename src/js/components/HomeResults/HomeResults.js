import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Button from '../Button/Button';
import './HomeResults.css';

const HomeResults = ({
  moviesFound,
  sortBy
}) => {
  const result = moviesFound > 1
    ? `${moviesFound} movies found`
    : `${moviesFound} movie found`;

  return (
    <Fragment>
      <div>{result}</div>
      <div className='rightSection'>
        Sort by
        <Button title='release date' className={sortBy === 'release_date' ? 'selected' : ''} onClick={() => console.log('click release date')} />
        <Button title='rating' className={sortBy === 'vote_average' ? 'selected' : ''} onClick={() => console.log('click rating')} />
      </div>
    </Fragment>
  )
}

HomeResults.propTypes = {
  moviesFound: PropTypes.number,
  sortBy: PropTypes.string
};

export default HomeResults;