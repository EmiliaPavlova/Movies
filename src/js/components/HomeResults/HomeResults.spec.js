import React from 'react';
import { mount } from 'enzyme';
import HomeResults from './HomeResults';
import './HomeResults.css';

const componentProps = (overrides = {}) => ({
  moviesFound: 2,
  sortBy: 'release_date',
  ...overrides
});

const mountHomeResults = props => mount(<HomeResults {...props} />);

describe('HomeResults', () => {
  describe('HomeResults attributes', () => {
    it('matches the snapshot for sort by release_date', () => {
      const props = componentProps();
      const wrapper = mountHomeResults(props);
      expect(wrapper).toMatchSnapshot('Sort by release_date');
    });
    it('matches the snapshot for sort by vote_average', () => {
      const props = componentProps({ sortBy: 'vote_average' });
      const wrapper = mountHomeResults(componentProps(props));
      expect(wrapper).toMatchSnapshot('Sort by vote_average');
    });
    it('matches the snapshot for only one movie found', () => {
      const props = componentProps({ moviesFound: 1 });
      const wrapper = mountHomeResults(componentProps(props));
      expect(wrapper).toMatchSnapshot('One movie found');
    });
  });
  describe('Event handlers', () => {
    it('calls onClick on click event', () => {
      const onClickMock = jest.fn();
      const props = componentProps({ onClick: onClickMock });
      const wrapper = mountHomeResults(props);
      const button = wrapper.find('button').at(0);
      button.simulate('click');
      // expect(onClickMock).toHaveBeenCalled();

      const actionButton = wrapper.find('button').at(1);
      actionButton.simulate('click');
      // expect(onClickMock).toHaveBeenCalled();
    });
  });
});
