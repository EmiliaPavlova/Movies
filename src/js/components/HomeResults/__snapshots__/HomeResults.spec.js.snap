// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeResults HomeResults attributes matches the snapshot for only one movie found: One movie found 1`] = `
<HomeResults
  moviesFound={1}
  sortBy="release_date"
>
  <div>
    1 movie found
  </div>
  <div
    className="rightSection"
  >
    Sort by
    <Button
      className="selected"
      onClick={[Function]}
      title="release date"
    >
      <button
        className="button selected "
        onClick={[Function]}
      >
        release date
      </button>
    </Button>
    <Button
      className=""
      onClick={[Function]}
      title="rating"
    >
      <button
        className="button  "
        onClick={[Function]}
      >
        rating
      </button>
    </Button>
  </div>
</HomeResults>
`;

exports[`HomeResults HomeResults attributes matches the snapshot for sort by release_date: Sort by release_date 1`] = `
<HomeResults
  moviesFound={2}
  sortBy="release_date"
>
  <div>
    2 movies found
  </div>
  <div
    className="rightSection"
  >
    Sort by
    <Button
      className="selected"
      onClick={[Function]}
      title="release date"
    >
      <button
        className="button selected "
        onClick={[Function]}
      >
        release date
      </button>
    </Button>
    <Button
      className=""
      onClick={[Function]}
      title="rating"
    >
      <button
        className="button  "
        onClick={[Function]}
      >
        rating
      </button>
    </Button>
  </div>
</HomeResults>
`;

exports[`HomeResults HomeResults attributes matches the snapshot for sort by vote_average: Sort by vote_average 1`] = `
<HomeResults
  moviesFound={2}
  sortBy="vote_average"
>
  <div>
    2 movies found
  </div>
  <div
    className="rightSection"
  >
    Sort by
    <Button
      className=""
      onClick={[Function]}
      title="release date"
    >
      <button
        className="button  "
        onClick={[Function]}
      >
        release date
      </button>
    </Button>
    <Button
      className="selected"
      onClick={[Function]}
      title="rating"
    >
      <button
        className="button selected "
        onClick={[Function]}
      >
        rating
      </button>
    </Button>
  </div>
</HomeResults>
`;
